k8s/scripts/sysbox-deploy-k8s.sh

```diff
@@ -948,6 +948,24 @@ function do_distro_adjustments() {
 	sed -i '/^kernel.unprivileged_userns_clone/ s/^#*/# /' ${sysbox_artifacts}/systemd/99-sysbox-sysctl.conf
 }
 
+# determines if running on a GKE cluster
+function check_gke() {
+	cluster_name=$(curl --connect-timeout 1 -s -H "Metadata-Flavor: Google" 169.254.169.254/computeMetadata/v1/instance/attributes/cluster-name || false)
```

Comment:
Curious, is this the GKE suggested way to find if the cluster is on GKE?

Reply:
@ctalledo it is possible not to enable Workload Identity on your GKE clusters if that's what you're getting at. This is a reliable way to tell if we are executing in GKE as long as WI is enabled. Another option is to look for fingerprint-like data on the GKE node's VM OS. However, the GKE docs warn against relying on the underlying OS not changing for your Daemonsets functionality to work correctly. GKE releases are decoupled from the node OS in my experience.

Another option might be to make this a bool that we flip as a config item so the GKE-specific functionality is explicitly enabled at install time and that could be covered by a documentation update?

Thank you for working on this @jamonation. I am no longer at the employer where I was impacted by this issue, but I can help with any GKE-specific validations or testing if you'd like an assist. Just let me know 👍🏽

Reply:
Hi @kevholmes, thanks for the response.

> This is a reliable way to tell if we are executing in GKE as long as WI is enabled.

Got it, thanks. I am not too familiar with WI yet. Is it enabled by default, and when would it make sense to disable it.

> Another option is to look for fingerprint-like data on the GKE node's VM OS. However, the GKE docs warn against relying on the underlying OS not changing for your Daemonsets functionality to work correctly. GKE releases are decoupled from the node OS in my experience.

Agreed.

> Another option might be to make this a bool that we flip as a config item so the GKE-specific functionality is explicitly enabled at install time and that could be covered by a documentation update?

In general we prefer to not add too many configs, unless absolutely required. In this case I think the solution you provide above might work just fine.

Reply:
@ctalledo: the only time I might disable WI support in a GKE cluster is in an environment where I'm executing workloads that I absolutely cannot trust. For instance, if I offered a CI/CD service that allowed folks external to my company to sign up with a credit card and run custom containers/code/cli commands in my k8s cluster. It sounds like a scenario where I might want Sysbox 😄. In a design like that, I would be:

a) disabling Workload Identity entirely 

or

b) creating NetworkPolicies to drastically limit who can speak to 169.254.169.254 / kube-system namespace / the gke-metadata-server pods themselves from within the k8s cluster(s) running these untrustworthy workloads.

The goal would be to ensure absolutely no private data could be leaked/exposed to bad actors from the Workload Identity service in GKE. This scenario of me blocking access to the metadata service IP could possibly cause comms issues if the NetworkPolicies were not crafted appropriately to take this requirement of being able to speak to the special IP / WI pods into account at the setup/install time of Sysbox via its daemonset.

Typically the Metadata service is "transparent" if we call it from a compute engine VM. However, in GKE, it's actually something we can see, restart, log, and instrument through tooling like Prometheus as it lives in the kube-system namespace.